source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -347,61 +384,112 @@ def __call__(
+            model = AutoModel.from_pretrained(f"theaiinstitute/{model_name}", trust_remote_code=True).eval()
+            return model.to(model_device)
 
-        if model_device is not None and self.model_zoo[model_name].device != model_device:
-            # want to offload vision model inference to another device
-            self.model_zoo[model_name] = self.model_zoo[model_name].to(model_device)
+        def _inference(model, images: torch.Tensor) -> torch.Tensor:
+            """Inference the Theia transformer model.
 
-        images = image(
-            env=env,
-            sensor_cfg=sensor_cfg,
-            data_type=data_type,
-            convert_perspective_to_orthogonal=convert_perspective_to_orthogonal,
-            normalize=True,  # want this for training stability
-        )
+            Args:
+                model: The Theia transformer model.
+                images: The preprocessed image tensor. Shape is (num_envs, height, width, channel).
+
+            Returns:
+                The extracted features tensor. Shape is (num_envs, feature_dim).
+            """
+            # Move the image to the model device
+            image_proc = images.to(model_device)
+            # permute the image to (num_envs, channel, height, width)
+            image_proc = image_proc.permute(0, 3, 1, 2).float() / 255.0
+            # Normalize the image
+            mean = torch.tensor([0.485, 0.456, 0.406], device=model_device).view(1, 3, 1, 1)
+            std = torch.tensor([0.229, 0.224, 0.225], device=model_device).view(1, 3, 1, 1)
+            image_proc = (image_proc - mean) / std
+
+            # Taken from Transformers; inference converted to be GPU only
+            features = model.backbone.model(pixel_values=image_proc, interpolate_pos_encoding=True)
+            return features.last_hidden_state[:, 1:]
 
-        image_device = images.device
+        # return the model, preprocess and inference functions
+        return {"model": _load_model, "inference": _inference}
 
-        if model_device is not None:
-            images = images.to(model_device)
+    def _prepare_resnet_model(self, model_name: str, model_device: str) -> dict:
+        """Prepare the ResNet model for inference.
 
-        proc_images = self.model_zoo_cfg[model_name]["preprocess"](images)
-        features = self.model_zoo_cfg[model_name]["inference"](self.model_zoo[model_name], proc_images)
+        Args:
+            model_name: The name of the ResNet model to prepare.
+            model_device: The device to store and infer the model on.
+
+        Returns:
+            A dictionary containing the model and inference functions.
+        """
+        from torchvision import models
 
-        return features.to(image_device).clone()
+        def _load_model() -> torch.nn.Module:
+            """Load the ResNet model."""
+            model = getattr(models, model_name)(weights="ResNet50_Weights.DEFAULT").eval()
```

Comment:
Should support other weights for other sizes as well like  18, 34, 101, etc

along lines of: 


```def create_resnet_model(resnet_name, weights):
    return {
        "model": lambda: getattr(models, resnet_name)(weights=weights).eval().to("cuda:0"),
        "preprocess": (
            lambda img: (
                (
                    img.permute(
                        0, 3, 1, 2
                    ).float()  # Convert [batch, height, width, 3] -> [batch, 3, height, width]
                    - torch.tensor([0.485, 0.456, 0.406], device=img.device).view(1, 3, 1, 1)
                )
                / torch.tensor([0.229, 0.224, 0.225], device=img.device).view(1, 3, 1, 1)
            )
            / 255
        ),
        "inference": lambda model, images: model(images),
    }

resnet_weights = {
    "resnet18": ResNet18_Weights.IMAGENET1K_V1,
    "resnet34": ResNet34_Weights.IMAGENET1K_V1,
    "resnet50": ResNet50_Weights.IMAGENET1K_V1,
    "resnet101": ResNet101_Weights.IMAGENET1K_V1
}

resnet_models = ["resnet18", "resnet34", "resnet50", "resnet101"]
for resnet_name in resnet_models:
    self.default_model_zoo_cfg[resnet_name] = create_resnet_model(resnet_name, resnet_weights[resnet_name])```